asapdiscovery-spectrum/asapdiscovery/spectrum/score_complex.py

```diff
@@ -0,0 +1,363 @@
+                df_save = pd.DataFrame(
+                    columns=[
+                        "premin-score-POSIT",
+                        "docking-score-POSIT",
+                        "computed-SchNet-pIC50",
+                        "computed-E3NN-pIC50",
+                        "computed-GAT-pIC50",
+                    ]
+                )
+
+        if ligand is not None:
+            df_save.insert(loc=0, column="lig-ID", value=ligand)
+        if prot_id is not None:
+            df_save.insert(loc=0, column="prot-ID", value=prot_id)
+        logging.debug("The DataFrame after docking is %s", df_save)
+
+        # Now save files for later scoring steps
+        docked_prepped = new_docking_dir / "prepped"
+        docked_prepped.mkdir(parents=True, exist_ok=True)
+        pdb_prep = docked_prepped / (aligned.stem + "_target.pdb")
+        sdf_ligand = docked_prepped / (aligned.stem + "_ligand.sdf")
+        # Really annoying, but Target and PreppedTarget have different functions for gen the PDB
+        if type(prepped_cmp) is Complex:
+            prepped_cmp.target.to_pdb(pdb_prep)
+        else:
+            prepped_cmp.target.to_pdb_file(pdb_prep)
+        ligand_pose.to_sdf(sdf_ligand)
+        logging.info(
+            "Saved prepped target as %s and ligand as %s",
+            pdb_prep.stem,
+            sdf_ligand.stem,
+        )
+
+        # RMSD score
+        logging.debug("The aligned file was saved in %s", aligned)
+        logging.info("Calculating RMSD of the ligand")
+        lig_rmsd = get_ligand_rmsd(
+            str(aligned), str(file_ref), True, rmsd_mode="oechem", overlay=False
+        )
+        df_save.insert(loc=len(df_save.columns), column="Lig-RMSD", value=lig_rmsd)
+        pout = f"Calculated RMSD of POSIT ligand pose = {lig_rmsd} with ref {file_ref.stem}"
+
+        if lig_rmsd < 0:
+            # Retry ligand rmsd with RDKit
+            logging.info("Trying Ligand RMSD on a different method")
+            sdf_ref = docked_prepped / (file_ref.stem + "_ligand.sdf")
+            lig_rmsd = get_ligand_rmsd(
+                str(aligned),
+                str(file_ref),
+                True,
+                rmsd_mode="rdkit",
+                overlay=False,
+                pathT=str(sdf_ligand),
+                pathR=str(sdf_ref),
+            )
+
+            if lig_rmsd < 0:
+                logging.warning("Retry ligand RMSD calculation failed.")
+
+        if inputs.bsite_rmsd:
+            logging.info("Calculating RMSD of the binding site")
+            try:
+                bsite_rmsd = get_binding_site_rmsd(
+                    aligned,
+                    file_ref,
+                   bsite_dist=5.0,
+                    ligres=inputs.lig_resname,
+                    chain_mob=inputs.dock_chain,
+                    chain_ref=inputs.ref_chain,
+                    rmsd_mode="heavy",
+                    aligned_temp=aligned,
+                )
+            except Exception as e:
```

Comment:
We should avoid catching generic exceptions. Any specific ones we should catch instead?